src/fsipy/simulations/simulation_common.py

```diff
@@ -0,0 +1,274 @@
+import json
+from typing import List, Union, Tuple
+from pathlib import Path
+
+import numpy as np
+from mpi4py import MPI as mpi
```

Comment:
Should we have mpi4py as a dependency?

Reply:
`mpi4py` is a dependency of Dolfin. so no need to add it explictly